stream/client.py

```diff
@@ -178,12 +178,16 @@ def create_user_token(self, user_id, **extra_data):
             payload[k] = v
         return jwt.encode(payload, self.api_secret, algorithm="HS256").decode("utf-8")
 
-    def create_jwt_token(self, resource, action, feed_id=None, user_id=None):
+    def create_jwt_token(self, resource, action, feed_id=None, user_id=None, **payload):
         """
         Setup the payload for the given resource, action, feed or user
         and encode it using jwt
         """
-        payload = {"action": action, "resource": resource}
+        if not payload:
+            payload = {"action": action, "resource": resource}
+        else:
+            payload["action"] = action
+            payload["resource"] = resource
```

Comment:
since we're on py3 now

`payload = {"action": action, "resource": resource, **params}` should be shorter

Reply:
That will override action from payload ISO the reverse. Since you have to explicitly state the action I think that should take precedence?

Reply:
it should fail in call side if there are multiple same keys, it wouldn't permit it.

we can have following too but shouldn't make a change:

`payload = {**params, "action": action, "resource": resource}`

Reply:
It does make a difference:

```
>>> params = {"action": "user-generated-garbage", }
>>> payload = {"action": "read", "resource": "personalization", **params}
>>> payload
{'action': 'user-generated-garbage', 'resource': 'personalization'}
>>> payload = {**params, "action": "read", "resource": "personalization"}
>>> payload
{'action': 'read', 'resource': 'personalization'}


And since the difference is subtle I like to spell it out. Explicit is better than implicit?

Reply:
Yes, that's right.

> it should fail in call side if there are multiple same keys, it wouldn't permit it.

This is for if it was a keyword argument which isn't. However, suggested change with order reversed is good.